src/features/defi/providers/fox-farming/components/FoxFarmingManager/Deposit/FoxFarmingDeposit.tsx

```diff
@@ -47,10 +45,7 @@ export const FoxFarmingDeposit: React.FC<FoxFarmingDepositProps> = ({
   const translate = useTranslate()
   const toast = useToast()
   const { query, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
-  const { assetNamespace, chainId, contractAddress, assetReference } = query
-
-  const assetId = toAssetId({ chainId, assetNamespace, assetReference })
-  const marketData = useAppSelector(state => selectMarketDataByAssetIdUserCurrency(state, assetId))
```

Comment:
oh boy what was this check even about 💀 market-data was always guaranteed to be defined, and this definitely proves it was a very bad idea for it to be!